Fix relative output dirs

rhpvorderman opened the pull request:
Due to some updates that got in cromwell version 49 the regex that creates the relative outputs was evaluated differently (correctly, to be precise). This made it so that any directory structure created by the user is lost. This broke all the testing on biowdl.
I have updated the test cases to notice such a regression in the future. I also fixed the regex to behave like it is described in the documentation.

cjllanwarne commented on engine/src/main/scala/cromwell/engine/workflow/lifecycle/finalization/CopyWorkflowOutputsActor.scala:
```diff
@@ -98,7 +98,7 @@ class CopyWorkflowOutputsActor(workflowId: WorkflowId, override val ioActor: Act
     // compiled for every single file.
     // "execution" should be optional, because its not created on AWS.
-    lazy val truncateRegex = ".*\\/call-.*\\/(execution\\/)?".r
+    lazy val truncateRegex = ".*/call-[^/]*/(execution/)?".r
```
NB this folder might also have `attempt-2` (or `-3`, ...) or now `cacheCopy` between the `call-...` and `/execution/`, so we'll probably want to take account of that too

cjllanwarne commented:
Thanks for finding and fixing this!

rhpvorderman commented:
I fixed the regex. It turned out that this also fixed any issues.

> Thanks for finding and fixing this!

Thank you for trusting me with push access on this repository. It makes it easier for me as all tests run immediately, also the tests that need private variables. Also I can restart jobs on travis now that looks like they are failed due to some intermittent connection error. I had to restart one for this PR, and it indeed turned green on the retry. This makes it easier for me to fix any bugs I find. The trust is much appreciated.
